### PR TITLE
KOGITO-5762 BDD: Cleanup crds

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-5762

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [ ] You've ran `make before-pr` and everything is working accordingly
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

File: hack/env.sh [hunk 1]
```diff
@@ -16,6 +16,8 @@
 CSV_DIR="config/manifests/bases"
 TEST_CONFIG_FILE="test/.default_config"
 
+DEPENDENT_CRDS_KEYS=(grafana hyperfoil infinispan kafka keycloak knative kogito mongodb prometheus)
```
Comment (sutaakar) [inline hack/env.sh:19]: Prometheus CRD shouldn't be deleted as it is used for cluster monitoring.

Comment (radtriste) [reply hack/env.sh]: Separated CRDS into sensitive and non-sensitive arrays

**Review by sutaakar — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**